### DISCUSSION TITLE
Change infill above arc overhang

### DISCUSSION THREAD

**Discussion by jbajcz** [nicolai-wachenschwan/arc-overhang-prusaslicer-integration#15] [Ideas]
I have had some very promising results after adding this to my prusaslicer. I was wondering if there were some other options when it comes to the geometry above the overhang. It appears that the hilbert curve is implemented for all geometry above that layer. Wouls it be possible to adjust the script so that all other layers are normal above the overhang? For example, the script only edits the layer where the overhang is, so then there are normal solid infill layers directly above that and then normal infill on top of those? 

In addition to that, is it possible to have the arc overhangs for internal bridging as well?

**Comment by nicolai-wachenschwan** [comment]
Hey,
the special hilbert-infill is needed to prevent warping. I testet the default solid infill with devastating results. Therefore I did not built a version to exclude hilbert generation, since it is kind of mandatory for print success.
If you want to avoid the generation anyway set the "specialCoolingZdist" to 0.

Arcs are only generated for overhangs when in contact with perimeter overhang. This avoids arc generation at unwanted spaces, e.g internally, since prusaslicer interprets the first solid infill as bridge infill, but there are not arcs needed. There are two possibilities to change this behavior:
1) make a little, steep chamfer close to the edge you want to fill with arcs. This will generate the necessary overhang perimeter. If needed adjust the MinBridgeLength
2) change the function "verifyinfillpolys" to your needs.

**Reply by jbajcz** [reply]
I have had some moderate success against warping using an objectively overpowered cooling setup (two 5045 blower-style fans) and just some minor tweaking to the settings. 

So the specialcoolingZdist is just the z height where it will make the Hilbert infill? After that, it will use the unedited gcode?

**Reply by nicolai-wachenschwan** [reply]
It will replace solid infill with hilbert and slow down all lines beeing close to the arcs. The cooling is reduced in these layers also. Alternatively you can chose to reduce cooling only for the areas near the arcs.
By setting it to 0 you avoid all special cooling settings.

**Reply by NateTG** [reply]
Depending on what it is that makes the Hilbert curve infill good for limited warping, it might make sense to try other "open" patterns like the octagram spiral or a bunch of disconnected ribs in a pattern like diamond plate.  (https://en.wikipedia.org/wiki/Tread_plate )

**Reply by nicolai-wachenschwan** [reply]
I put a lot on thought into a well suited pattern. It needs to have short lines to keep the thermal contraction small. But the lines also need to be printed long/quasi continuous, else it wont be efficient+ to many extrusion stops, which might clog the nozzle. Also every line start/stop can induce defects. 
Research from Catchpole-Smith et al. (https://www.researchgate.net/publication/313685481_Fractal_Scan_Strategies_for_Selective_Laser_Melting_of_'Unweldable'_Nickel_Superalloys) shows unsuprisingly reduced stress for the hilbert curve, which is also easy to implement. so it seemed like a good choice.

But you can implement your own patterns and give it a shot, thats the beauty of the python script :)

**Comment by dining-philosopher** [comment]
Hello! Can i turn off the hilbert curve or reduce the numbers of hilbert layers? The current version makes hilbert curves even at the top-facing layers, i. e. after the infill where obviously no warping reduction is needed (and also makes severe spaghetti mess at these layers when actually printing). I looked the parameter list in the python file and could not find any similar parameter.

**Comment by nicolai-wachenschwan** [comment]
Yes: Simply put special cooling Z dist to 0
very unfortunate to hear of the top layer problem… should have been fixed with the Minimum Bridging length Parameter…
